File: ai-llm-connector/backend-java/src/main/java/com/angular/ai/controller/LlmController.java
package com.angular.ai.controller;

import com.angular.ai.config.AiServicesConfig;
import com.angular.ai.service.llm.ChatGptService;
import com.angular.ai.service.llm.ClaudeService;
import com.angular.ai.mock.llm.ChatGptMock;
import com.angular.ai.mock.llm.ClaudeMock;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/llm")
public class LlmController {

    @Value("${use.mock:false}")
    private boolean useMock;
    
    private final ChatGptService chatGptService;
    private final ClaudeService claudeService;
    private final ChatGptMock chatGptMock;
    private final ClaudeMock claudeMock;

    public LlmController(ChatGptService chatGptService, ClaudeService claudeService,
                         ChatGptMock chatGptMock, ClaudeMock claudeMock) {
        this.chatGptService = chatGptService;
        this.claudeService = claudeService;
        this.chatGptMock = chatGptMock;
        this.claudeMock = claudeMock;
    }
    
    @PostMapping("/{type}/{llm}")
    public Map<String, Object> handleRequest(@PathVariable String type,
                                             @PathVariable String llm,
                                             @RequestBody(required = false) Map<String, Object> input) {
        System.out.println("POST /api/llm/" + type + "/" + llm);
        System.out.println("▶️ useMock = " + useMock);
        System.out.println("▶️ payload = " + input);

        if (input == null || input.isEmpty()) {
            return Map.of(
                "success", false,
                "data", "missing or empty request body"
            );
        }

        String llmKey = llm.toLowerCase();
        String result;

        try {
            switch (llmKey) {
                case "chatgpt":
                    result = useMock ? chatGptMock.reply(type, input) : chatGptService.reply(type, input);
                    break;
                case "claude":
                    result = useMock ? claudeMock.reply(type, input) : claudeService.reply(type, input);
                    break;
                default:
                    return Map.of(
                        "success", false,
                        "data", "unknown-provider: " + llm
                    );
            }

            return Map.of(
                "success", true,
                "type", type,
                "llm", llmKey,
                "data", result
            );

        } catch (Exception e) {
            System.err.println("❌ Error while handling request: " + e.getMessage());
            return Map.of(
                "success", false,
                "data", "internal-error",
                "error", e.getMessage()
            );
        }
    }

    
    
}

File: ai-llm-connector/backend-javascript/src/controllers/llm/llm.controller.js
import chatgptMock from '../../mocks/llm/chatgpt.mock.js';
import claudeMock from '../../mocks/llm/claude.mock.js';
import chatgptService from '../../services/llm/chatgpt.service.js';
import claudeService from '../../services/llm/claude.service.js';

const useMock = process.env.USE_MOCK === 'true';

function isUnauthorizedError(message) {
  return message.includes('unauthorized') || message.includes('401');
}

function getProvider(llm) {
  const providers = {
    chatgpt: {
      mock: chatgptMock,
      real: chatgptService,
    },
    claude: {
      mock: claudeMock,
      real: claudeService,
    },
  };

  return providers[llm] || null;
}

async function callLLM(type, llm, data) {
  try {
    const provider = getProvider(llm);
    if (!provider) {
      return { error: 'unknown-provider' };
    }

    const handlerFunction = useMock ? provider.mock : provider.real;
    const result = await handlerFunction(type, data);

    return { data: result };
  } catch (err) {
    console.error('❌ callLLM error:', err);

    return { error: 'internal-error' };
  }
}

export async function handleLlmRequest(req, res) {
  const { type, llm } = req.params;
  const input = req.body;

  try {
    const { data, error } = await callLLM(type, llm, input);

    if (error) {
      return res.status(400).json({ success: false, data: error });
    }

    return res.json({ success: true, data: data });
  } catch (err) {
    console.error('❌ Erreur serveur :', err.message);
    const msg = err.message?.toLowerCase() || '';
    const errorText = isUnauthorizedError(msg) ? 'unauthorized API KEY' : 'internal-error';

    return res.status(500).json({ success: false, data: errorText });
  }
}
